### PR TITLE
added sampled_out dmarc report

### PR DESCRIPTION
**Pull request by cpragadeesh**
Also applied next policy by severity in case of sampled_out emails as per [RFC7489](https://tools.ietf.org/html/rfc7489#section-6.6.4).

### CODE REVIEW

File: src/plugins/lua/dmarc.lua [hunk 1]
```diff
@@ -321,11 +321,16 @@ local function dmarc_callback(task)
         if not pct or pct == 100 or (math.random(100) <= pct) then
           task:insert_result(dmarc_symbols['quarantine'], res, lookup_domain .. ' : ' .. reason_str, dmarc_policy)
           disposition = "quarantine"
+        else
+          task:insert_result(dmarc_symbols['softfail'], res, lookup_domain .. ' : ' .. reason_str, "none", "sampled_out")
```
Comment (fatalbanana) [inline src/plugins/lua/dmarc.lua]: You'll want to set `disposition = 'none'` here

**Comment by fatalbanana** [comment]
That's a good start, thanks, but this information is still missing from DMARC reporting information in Redis.

File: src/plugins/lua/dmarc.lua (hunk 1 above)
Comment (cpragadeesh) [inline src/plugins/lua/dmarc.lua]: disposition is set to 'none' initially here:
`local disposition = 'none'`

Should I set it to 'none' again regardless?

Comment (fatalbanana) [inline src/plugins/lua/dmarc.lua]: No, sorry. :) It should be fine in fact.

Comment (fatalbanana) [inline src/plugins/lua/dmarc.lua]: Oh, "none" here should be dmarc_policy.

File: src/plugins/lua/dmarc.lua [hunk 2]
```diff
@@ -321,11 +321,16 @@ local function dmarc_callback(task)
...
         end
       elseif dmarc_policy == 'reject' then
         if not pct or pct == 100 or (math.random(100) <= pct) then
           task:insert_result(dmarc_symbols['reject'], res, lookup_domain .. ' : ' .. reason_str, dmarc_policy)
           disposition = "reject"
+        else
+          task:insert_result(dmarc_symbols['quarantine'], res, lookup_domain .. ' : ' .. reason_str, "quarantine", "sampled_out")
```
Comment (fatalbanana) [inline src/plugins/lua/dmarc.lua]: "quarantine" should be dmarc_policy.

**Comment by cpragadeesh** [comment]
@fatalbanana Hope I got it right this time :)

**Pull request merged**